### ISSUE TITLE
Log client errors with Tracks

### ISSUE THREAD

**Issue by oaratovskyi** [Automattic/woocommerce-payments#5855]
### Description
This is a follow-up issue after #5779. We agreed to log client errors properly in this one. As an idea, we could do it with Tracks, see the example here p1679656508824199/1679655227.304219-slack-C03KTTK2YMA. But we could also do it differently. The main goal is to track errors from `isEligibleForPo()` so we know when things go wrong.

### Dev Notes
We should move the try/catch to `isEligibleForPo`. Also, decide if we want to move it somewhere else. 

### Additional context
Slack conversation: p1679655227304219-slack-C03KTTK2YMA

**Comment by oaratovskyi** [comment]
After exploration and RFC outcome paJDYF-cEM-p2#comment-23398 I closed the connected PR.
TL;DR Tracks is not the best tool for this, we should explore more options. 
Closing this issue and will create another one - more general.

**Comment by oaratovskyi** [comment]
issue created for further research https://github.com/Automattic/woocommerce-payments/issues/8272
